packages/core/src/Table/TableCell.js

```diff
@@ -1,69 +1,129 @@
-}) => (
-    <td
-        colSpan={colSpan}
-        rowSpan={rowSpan}
-        className={cx({ dense }, className)}
-        data-test={dataTest}
-    >
-        {children}
-
-        <style jsx>{tableCellStyles}</style>
-    </td>
+export const TableCell = forwardRef(
+    (
+        {
+            active,
+            align,
+            bordered,
+            children,
+            className,
+            colSpan,
+            dataTest,
+            error,
+            fixed,
+            large,
+            left,
+            muted,
+            rowSpan,
+            scope,
+            tag,
+            valid,
+            width,
+            onClick,
+        },
+        ref
+    ) => {
+        const CellTag = tag || fixed ? 'th' : 'td'
+        return (
+            <CellTag
+                ref={ref}
+                colSpan={colSpan}
+                rowSpan={rowSpan}
+                onClick={onClick}
+                className={cx(className, {
+                    active,
+                    bordered,
+                    error,
+                    fixed,
+                    large,
+                    muted,
+                    valid,
+                })}
+                data-test={dataTest}
+                scope={scope}
+            >
+                {children}
+                <style jsx>{styles}</style>
+                <style jsx>{`
+                    th,
+                    td {
+                        left: ${left};
+                        text-align: ${align};
+                        width: ${width};
+                    }
+                `}</style>
+            </CellTag>
+        )
+    }
 )
 
+TableCell.displayName = 'TableCell'
+
 TableCell.defaultProps = {
+    align: 'left',
     dataTest: 'dhis2-uicore-tablecell',
+    left: 'auto',
+    width: 'auto',
 }
 
+const stylePropType = propTypes.mutuallyExclusive(
+    ['valid', 'error', 'muted'],
+    propTypes.bool
+)
+const requiredIfFixedPropType = propTypes.requiredIf(
+    props => props.fixed,
+    propTypes.string
+)
 /**
  * @typedef {Object} PropTypes
  * @static
+ * @prop {node} [children]
+ * @prop {string} [className]
+ * @prop {string} [dataTest=dhis2-uicore-tablecell]
  * @prop {string} [colSpan]
  * @prop {string} [rowSpan]
- * @prop {bool} [dense]
- * @prop {Node} [children]
- * @prop {string} [className]
- * @prop {string} [dataTest]
+ * @prop {string} [scope]
+ * @prop {string} [tag] HTML tag to render (will default to td)
+ * @prop {boolean} [active] To toggle background color, for example for editing
+ * @prop {function} [onClick]
+ * @prop {boolean} [large]
+ * @prop {boolean} [bordered]
+ * @prop {boolean} [fixed]
+ * @prop {string} [width=auto] Required when fixed
+ * @prop {string} [left=auto] Required when fixed
+ * @prop {left|center|right} [align=left]
+ * @prop {boolean} [error] Mutually exclusive
+ * @prop {boolean} [valid] Mutually exclusive
+ * @prop {boolean} [muted] Mutually exclusive
  */
 TableCell.propTypes = {
```

Comment:
The `bordered` and `large` are technically props that apply to a cell because they impact the `td` style. 

However, the design system dictates that all cells in a table should be homogenous regarding these styles and as such you could argue these props should have been part of the `Table`. 

So there is some tension there. In our previous implementation we used context to solve this puzzle (we had `suppressZebraStriping` then instead of `large` and `bordered`). The `Table` used to accept a  `suppressZebraStriping` prop and exposed that props via a context provider to all its children. The actual styles were still defined in the `TableCell`, so as whole this seemed to make quite a lot of sense....

However, we've since realised that using context in a library unwillingly turns that library into a singleton, so we decided to move away from using context.

Another way to keep these props on the `Table` and still make sure the `TableCell` knows about these props is to use `React.cloneElement()`. We've done so extensively in the past and we aren't too happy about this solution. Furthermore, in this case we'd actually be passing elements down quite a few levels in the component hierarchy: `table > tbody > tr > td`. That would make this solution even less nice in this case.

Yet another way to keep these props on the `Table`, would be to declare `td` styles in the `Table` component. But this would break the style encapsulation principle.

I discussed this with @ismay at some point, and as he pointed out: normally in an app we will be mapping over data when producing a table anyway, so setting a few props on a cell isn't going to be that inconvenient. I think he's correct and that's why in the end i decided to keep these props on the `TableCell` and `TableCellHead` instead of on the `Table`.